### PR TITLE
Add the ability to store application data

### PR DESCRIPTION
**Pull request by ajaxscape**

### CODE REVIEW

File: src/persistence/entities/applicationData.entity.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+class ApplicationData extends BaseModel {
+  static get dynamicsEntity () {
+    return 'defra_webdata'
+  }
+
+  static get mapping () {
+    return [
+      { field: 'id', dynamics: 'defra_webdataid' },
+      { field: 'applicationId', dynamics: '_defra_applicationid_value', bind: { id: 'defra_applicationId', relationship: 'defra_defra_application_defra_webdata', dynamicsEntity: 'defra_applications' } },
+      { field: 'data', dynamics: 'defra_data', encode: true }
```
Comment (bensagar-ea) [inline src/persistence/entities/applicationData.entity.js]: The encode value is only used for filtering and I don't think we'd ever want to filter by this value

Comment (bensagar-ea) [inline src/persistence/entities/applicationData.entity.js]: This field could have a max length of 20,000 defined to prevent accidental attempts at insertion of arbitrary strings.

File: src/persistence/entities/applicationData.entity.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+'use strict'
+
+const BaseModel = require('./base.entity')
```
Comment (bensagar-ea) [inline src/persistence/entities/applicationData.entity.js]: Should refer to this as BaseEntity, just to avoid confusion.

File: test/persistence/entities/applicationData.entity.test.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+})
+
+lab.experiment('ApplicationData Model tests:', () => {
+  lab.test('getById() method correctly retrieves an ApplicationData object', async () => {
+    const spy = sinon.spy(DynamicsDalService.prototype, 'search')
+    const applicationData = await ApplicationData.getById(context, applicationDataId)
+    Code.expect(spy.callCount).to.equal(1)
+    Code.expect(applicationData.id).to.equal(fakeApplicationData.id)
+    Code.expect(applicationData.applicationId).to.equal(fakeApplicationData.applicationId)
+    Code.expect(applicationData.data).to.equal(JSON.stringify(fakeApplicationData.data))
```
Comment (bensagar-ea) [inline test/persistence/entities/applicationData.entity.test.js]: I wonder if there any obscure edge cases that we may need to test for here as the data field is being JSON.stringify'd twice?

I can't think of any off the top of my head but there might be something.

File: src/models/dataStore.model.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+const BaseModel = require('./base.model')
+const ApplicationData = require('../persistence/entities/applicationData.entity')
+
+module.exports = class DataStore extends BaseModel {
```
Comment (bensagar-ea) [inline src/models/dataStore.model.js]: This class doesn't really seem to be modelling anything in the business layer and is pretty much just a wrapper around the the underlying entity.

Is it needed at this point or could we just create business models as required that use the entity?

File: src/persistence/entities/applicationData.entity.js (hunk 1 above)
Comment (ajaxscape) [reply src/persistence/entities/applicationData.entity.js]: Removed

Comment (ajaxscape) [reply src/persistence/entities/applicationData.entity.js]: Added max length

File: src/persistence/entities/applicationData.entity.js (hunk 2 above)
Comment (ajaxscape) [reply src/persistence/entities/applicationData.entity.js]: Changed to BaseEntity where required

File: src/models/dataStore.model.js (hunk 4 above)
Comment (ajaxscape) [reply src/models/dataStore.model.js]: We'll look into this later
